packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -69,104 +53,20 @@ export function useMutateQueueEntries() {
-  useEffect(() => {
-    const isSearchCriteriaUpdated = !isEqual(currentSearchCriteria, searchCriteria);
-    const isRepUpdated = currentRep !== rep;
-    if (isSearchCriteriaUpdated || isRepUpdated) {
-      if (isSearchCriteriaUpdated) {
-        setCurrentSearchCriteria(searchCriteria);
-      }
-      if (isRepUpdated) {
-        setCurrentRep(rep);
-      }
-      refetchAllData(rep, searchCriteria);
-    }
-  }, [currentRep, currentSearchCriteria, refetchAllData, rep, searchCriteria, setCurrentSearchCriteria]);
-
-  const { data: pageData, isValidating, error: pageError } = useSWR<QueueEntryResponse, Error>(pageUrl, openmrsFetch);
+  });
 
   useEffect(() => {
-    const nextUrl = getNextUrlFromResponse(pageData);
-    const stillWaitingForMutate = waitingForMutate && !isValidating;
-    if (waitingForMutate && isValidating) {
-      setWaitingForMutate(false);
-    }
-    if (pageData && !isValidating && !stillWaitingForMutate) {
-      // We've got results! Time to update the data array and move on to the next page.
-      if (pageData?.data?.totalCount > -1 && pageData?.data?.totalCount !== totalCount) {
-        setTotalCount(pageData?.data?.totalCount);
-      }
-      if (pageData?.data?.results) {
-        const newData = [...data];
-        newData[currentPage] = pageData?.data?.results;
-        setData(newData);
-      }
-      setCurrentPage(currentPage + 1);
-      setPageUrl(nextUrl);
-      // If we're mutating existing data, then we again need to wait for the mutate to work,
-      // since useSWR will (again) first return stale data with isValidating set to false.
-      const inMutateMode = data.length > currentPage;
-      if (inMutateMode && nextUrl) {
-        setWaitingForMutate(true);
-      }
-    }
-    // It may happen that there are fewer pages in the new data than in the old data. In this
-    // case, we need to remove the extra pages, which are stored on the `data` array.
-    // Note that since we mutated the `data` state earlier in this function, it is important to
-    // use the functional form of `setData` so as not to use the stale `data` state.
-    if (!nextUrl) {
-      // I will not be very suprised if there is an off-by-one error here.
-      if (data.length > currentPage + 1) {
-        setData((prevData) => {
-          const newData = [...prevData];
-          newData.splice(currentPage + 1);
-          return newData;
-        });
-      }
-    }
-  }, [pageData, data, currentPage, totalCount, waitingForMutate, isValidating]);
-
-  useEffect(() => {
-    // An error to one is an error to all
-    if (pageError) {
-      setError(pageError);
-    }
-  }, [pageError]);
+    setPageUrl(getInitialUrl(rep, searchCriteria));
+  }, [searchCriteria, rep]);
 
-  const queueUpdateListener = useCallback(() => {
-    refetchAllData();
-  }, [refetchAllData]);
+ const queueUpdateListener = useCallback(() => {
```

Comment:
I think we simplify this further. I think we can get rid of all the `queueUpdateListener` and `queue-entry-updated` stuff, and whatever function it is that triggers the `queue-entry-updated` event. We can simply expose the mutate function and use that.